fempy/FitFunctions.cxx

```diff
@@ -0,0 +1,215 @@
+    TSpline5* sp5 = new TSpline5("sp5", xKnots, yKnots, numKnots, "");
+    return sp5->Eval(x[0]);
+}
+
+double Spline3Range(double *x, double *par){
+    int numKnots = 10;
+    Double_t xKnots[numKnots];
+    Double_t yKnots[numKnots];
+    for(int iKnot=0; iKnot<numKnots; iKnot++){
+        xKnots[iKnot] = par[iKnot];
+        yKnots[iKnot] = par[numKnots+iKnot];
+    }
+    TSpline3* sp3 = new TSpline3("sp3", xKnots, yKnots, numKnots, "");
+    
+    //if(x[0]<95 || x[0]>) return 0;
+    //else {
+    //    return sp3->Eval(x[0]);
+    //}
+    return sp3->Eval(x[0]);
+}
+
+double Spline3Histo(double *x, double *par){
+
+    TFile *histoFile = TFile::Open("/home/mdicostanzo/an/LPi/Analysis/SimAllMothersMerged.root", "r");
+    TH1D *splineHisto = static_cast<TH1D*>(histoFile->Get("Pairs/hSE_2113122_NoDirectSigmaXi_smearednew"));
+    TSpline3* sp3 = new TSpline3(splineHisto);
+    
+    return sp3->Eval(x[0]);
+
+}
+
+ double Landau(double *x, double *par)
+ {
+    // par[0]: norm
+    // par[1]: mean
+    // par[2]: sigma
+
+    //if (par[2] <= 0) return 0;
+    //double den = ::ROOT::Math::landau_pdf( (x[0]-par[1])/par[2] );
+    //if (!par[0]) return den;
+    //return den/par[2];
+ 
+    if (par[2] <= 0) return 0;
+    double den = ::ROOT::Math::landau_pdf( (x[0]-par[1])/par[2] );
+    if (!par[0]) return den;
+    return par[0]*den;
+ 
+}
+
+//Double_t Sill_WithPS(Double_t *x, Double_t *par) {
+//
+//  // x[0]: k*
+//  // par: [0] "normalisation" constant
+//  //      [1] width
+//  //      [2] mass
+//  //      [3] resonance pT
+//  //      [4] kinetic decoupling temperature
+//
+//  Double_t LambdaFactor = 290;
+//
+//  if (x[0] < 0)
+//    return 0;
+//
+//  // double t = x[0];
+//
+//  Double_t kstar = x[0];
+//  Double_t Thresh = MassPion + MassProton;
+//
+//  double MotherMass = sqrt(kstar * kstar + MassPion * MassPion) + sqrt(kstar * kstar + MassProton * MassProton);
+//
+//  if (MotherMass < Thresh)
+//    return 0;
+//
+//  Double_t Width = par[1] * par[2] / sqrt(par[2] * par[2] - Thresh * Thresh);
+//  Double_t arg0 = 2 * MotherMass / TMath::Pi();
+//  Double_t arg1 = sqrt(MotherMass * MotherMass - Thresh * Thresh) * Width;
+//  Double_t arg2 = pow(MotherMass * MotherMass - par[2] * par[2], 2.);
+//  Double_t arg3 = pow(sqrt(MotherMass * MotherMass - Thresh * Thresh) * Width, 2.);
+//
+//  Double_t ResonancePt = par[3];
+//  Double_t Temperature = par[4]; // orignal paper 160
+//  Double_t PhaseSpace = (MotherMass / sqrt(MotherMass * MotherMass + ResonancePt * ResonancePt)) * exp(-sqrt(MotherMass * MotherMass + ResonancePt * ResonancePt) / Temperature);
+//  Double_t PhaseSpaceNorm = exp(-sqrt(Thresh * Thresh + ResonancePt * ResonancePt) / Temperature) * Temperature;
+//
+//  // return (par[0]*arg0*arg1/(arg2 + arg3))*PhaseSpace/(PhaseSpaceNorm*ME_Value); // *abs(Derivative_dM_dkStar(kstar));
+//
+//  return (par[0] * arg0 * arg1 / (arg2 + arg3)) * PhaseSpace / (PhaseSpaceNorm)*abs(Derivative_dM_dkStar(kstar));
+//
+//}
+
```

Comment:
For me it's fine to have the Sill in the code, any reason why it's commented out?